content/documentation/pages/5-feature-guides/2-batch/6-composed-task.md

```diff
@@ -0,0 +1,643 @@
+
+In this example we see that the split1-3 were fired simultaneously before CTR launched our transition app. And we added a new argument `--split-thread-core-pool-size=4` this basically states that the composed task runner can run 4 apps simultaneously.
+
+#### What did I just do?
+
+Again, what is all that stuff I put in the command line? So for this example we wanted to show how to use both command line args and properties. We used the arguments to establish the properties for the `Composed Task Runner`:
+
+1. `--increment-instance-enabled=true` states that we want to be able to execute this Composed Task multiple times. (Composed Tasks are build using Spring Batch and thus are batch jobs)
+1. `--interval-time-between-checks=1000` states that the `Composed Task Runner` will wait 1 second between checks to make sure that a task is complete (the default is 10 seconds).
+1. `--split-thread-core-pool-size=4` states that we want up to 4 simulatenous tasks to run at the same time.
+1. `--spring.cloud.task.closecontextEnabled=true` states that we want the Spring Context to close when the `Composed Task Runner`.
+
+[[note]]
+| When using `split` you must set the `spring.cloud.task.closecontextEnabled` property as shown above.
+
+We used the properties to set the transition-sample properties. Notice we had 4 sections for the taskapp.exitMessage property lets breakdown the `app.basictransition.transition-sample.taskapp.exitMessage=FAILED` into its components:
+
+1. `app` - tells spring cloud dataflow that this property is to be used for task application.
+1. `basictransition` - tells dataflow that this property will be used by the `Composed Task Runner` or one of the apps in the graph
+1. `transition-sample` - tells dataflow that this property is to be used by the transition-sample
+1. `transition-sample.exitMessage` - the property itself.
+
+More can be read about the sections of a property and the different property types [here](#passing-properties)
+
+#### Configuring your split
+
+In the example above we configured the behavior of our split in the composed task by using the `spring.cloud.task.closecontextEnabled` and `split-thread-core-pool-size` properties. Let's look at all the properties you can take advantage of when using splits.
+
+- spring.cloud.task.closecontextEnabled - When using splits, this property is required to be set to `true` because the context will not close, because threads were allocated to support the split.
+- split-thread-core-pool-size - Establishes the initial number of threads required for the splits in the composed task. Each child task contained in a split requires a thread in order to execute. (Defaults to 1)
+- split-thread-max-pool-size - The maximum number threads to be allocated.
+- split-thread-queue-capacity - The number of tasks that should be enqueued if all threads are in use before a new thread is allocated.
+
+##### Basic Split Sizing
```

Comment:
I think as an improvement we can do this counting ourselves and set it to the appropriate value, in particular for cf/k8s platforms.  If it goes above some threshold, we can reject the execution since then the split wouldn't really be a split.